src/overlays/actors/ovl_Bg_Haka_Bombwall/z_bg_haka_bombwall.c

```diff
@@ -75,76 +75,84 @@ s32 func_80BD5E00(BgHakaBombwall* this) {
 }
 
-#ifdef NON_MATCHING
 void func_80BD5E6C(BgHakaBombwall* this, PlayState* play) {
-    s32 i;
-    Vec3f spE0;
-    Vec3f spD4;
-    Vec3f spC8;
-    Vec3f spBC;
-    f32 temp_fs0;
-    f32 phi_fs1;
+    u32 i;
+    Vec3f mtxPos;
+    Vec3f mtxVel;
+    Vec3f pos;
+    Vec3f vel;
+    f32 offsetPosX;
+    f32 offsetPosY;
+    s16 scale;
     s16 phi_s0;
-    s16 gravity;
     s16 phi_t0;
+    s16 gravity;
 
     Matrix_Push();
     Matrix_RotateYS(this->dyna.actor.shape.rot.y, MTXMODE_NEW);
-    temp_fs0 = 0.0f;
-    phi_fs1 = 0.0f;
+
+    offsetPosX = 0.0f;
+    offsetPosY = 0.0f;
 
     for (i = 0; i < 30; i++) {
-        temp_fs0 += 60.0f + (Rand_ZeroOne() * 20.0f);
-        if (temp_fs0 > 75.0f) {
-            temp_fs0 -= 150.0f;
+        offsetPosX += 60.0f + (Rand_ZeroOne() * 20.0f);
+
+        if (offsetPosX > 75.0f) {
+            offsetPosX -= 150.0f;
         }
-        spC8.x = temp_fs0;
 
-        phi_fs1 += 5;
-        spC8.y = phi_fs1;
+        offsetPosY += 5;
 
-        spC8.z = (Rand_ZeroOne() * 20.0f) - 10.0f;
+        pos.x = offsetPosX;
+        pos.y = offsetPosY;
+        pos.z = (Rand_ZeroOne() * 20.0f) - 10.0f;
 
-        spBC.x = ((Rand_ZeroOne() - 0.5f) * 5.0f) + (temp_fs0 * (4.0f / 75.0f));
-        spBC.y = (Rand_ZeroOne() * 7.0f) - 2.0f;
-        spBC.z = (Rand_ZeroOne() * 4.0f) - 2.0f;
+        vel.x = ((Rand_ZeroOne() - 0.5f) * 5.0f) + (offsetPosX * (4.0f / 75.0f));
+        vel.y = (Rand_ZeroOne() * 7.0f) - 2.0f;
+        vel.z = (Rand_ZeroOne() * 4.0f) - 2.0f;
 
-        Matrix_MultVec3f(&spC8, &spE0);
-        Matrix_MultVec3f(&spBC, &spD4);
+        Matrix_MultVec3f(&pos, &mtxPos);
+        Matrix_MultVec3f(&vel, &mtxVel);
 
-        spE0.x += this->dyna.actor.world.pos.x;
-        spE0.y += this->dyna.actor.world.pos.y;
-        spE0.z += this->dyna.actor.world.pos.z;
+        mtxPos.x += this->dyna.actor.world.pos.x;
+        mtxPos.y += this->dyna.actor.world.pos.y;
+        mtxPos.z += this->dyna.actor.world.pos.z;
+
+        //! FAKE
+
+        if (1) {}
 
         if ((i & 3) == 0) {
             phi_s0 = 32;
-            func_800BBFB0(play, &spE0, 60.0f, 2, 100, 120, 1);
+            func_800BBFB0(play, &mtxPos, 60.0f, 2, 100, 120, 1);
         } else {
             phi_s0 = 64;
         }
 
-        if (i % 2 != 0) {
+        if ((i % 2) != 0) {
             phi_s0 |= 1;
             phi_t0 = 1;
         } else {
             phi_t0 = 0;
         }
 
-        if (D_80BD64CC[i & 3] >= 16) {
+        scale = sRockScale[i & 3];
+
+        //! FAKE
+    fake_label:;
+
+        if (scale >= 16) {
             gravity = -550;
         } else {
             gravity = -450;
         }
 
-        EffectSsKakera_Spawn(play, &spE0, &spD4, &spE0, gravity, phi_s0, 30, 0, 0, D_80BD64CC[i & 3], phi_t0, 0, 50, -1,
-                             OBJECT_IKANA_OBJ, object_haka_obj_DL_001680);
+        EffectSsKakera_Spawn(play, &mtxPos, &mtxVel, &mtxPos, gravity, phi_s0, 30, 0, 0, scale, phi_t0, 0, 50, -1,
```

Comment:
It seems better if these were the non-prefixed `pos` and `vel`, since they are the actual position and velocity the effect uses. I'm not sure what to name the other variables, but will think on it.

Reply:
Looking it over again, I would think `offsetPos` and `offsetVel` would work.

Reply:
so what should i call the original pos and vel then? i sure can't call them mtxPos and mtxVel